c_emulator/riscv_sim.c

```diff
@@ -1088,15 +1089,23 @@ int main(int argc, char **argv)
     }
     printf("Connected\n");
   } else
-    entry = load_sail(file);
+    entry = load_sail(file0);
 #else
-  uint64_t entry = load_sail(file);
+  uint64_t entry = load_sail(file0);
 #endif
+  /* Load any additional ELF files into memory */
+  for (int i = files_start + 1; i < argc; i++) {
+    bool is32bit;
+    uint64_t entry;
+    fprintf(stdout, "Loading additional ELF file %s.\n", argv[i]);
+    load_elf(argv[i], &is32bit, &entry);
+    check_elf(is32bit);
+  }
 
   /* initialize spike before sail so that we can access the device-tree blob,
    * until we roll our own.
    */
-  init_spike(file, entry, rv_ram_size);
+  init_spike(file0, entry, rv_ram_size);
```

Comment:
Does Spike have the ability to load multiple .elf files?   If so,  shouldn't this also be put into a "for" loop?

Reply:
I have never used the init_spike logic, we may need to call `tv_load_elf(s, f);` for each of the files. Maybe @pmundkur can comment on this.

Reply:
We should not slow this down for Spike.